### PR TITLE
services.mailman: Give mailman a posix group

### PR DESCRIPTION
**Pull request by benley**
###### Motivation for this change

This greatly simplifies things like permitting nginx to read `/var/lib/mailman-web/static`, or non-root postfix to read `/var/lib/mailman/data`

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/mail/mailman.nix [hunk 1]
```diff
@@ -241,8 +246,8 @@ in {
         mailmanCfg=$mailmanDir/mailman-hyperkitty.cfg
         mailmanWebCfg=$mailmanWebDir/settings_local.json
 
-        install -m 0700 -o mailman -g nogroup -d $mailmanDir
-        install -m 0700 -o ${cfg.webUser} -g nogroup -d $mailmanWebDir
+        install -m 0700 -o mailman -g mailman -d $mailmanDir
+        install -m 0700 -o ${cfg.webUser} -g mailman -d $mailmanWebDir
```
Comment (alyssais) [inline nixos/modules/services/mail/mailman.nix]: I don’t understand the purpose of this change. The directories are not group-readable.

File: nixos/modules/services/mail/mailman.nix [hunk 2]
```diff
@@ -253,14 +258,14 @@ in {
                 --arg archiver_key "$hyperkittyApiKey" \
                 --arg secret_key "$secretKey" \
                 >"$mailmanWebCfgTmp"
-            chown ${cfg.webUser} "$mailmanWebCfgTmp"
+            chown ${cfg.webUser}:mailman "$mailmanWebCfgTmp"
```
Comment (alyssais) [inline nixos/modules/services/mail/mailman.nix]: The config contains secrets. What reason would there be for another service to want to read this file?

File: nixos/modules/services/mail/mailman.nix [hunk 3]
```diff
@@ -253,14 +258,14 @@ in {
...
             mv -n "$mailmanWebCfgTmp" $mailmanWebCfg
         fi
 
         hyperkittyApiKey="$(jq -r .MAILMAN_ARCHIVER_KEY $mailmanWebCfg)"
         mailmanCfgTmp=$(mktemp)
         sed "s/@API_KEY@/$hyperkittyApiKey/g" ${mailmanHyperkittyCfg} >"$mailmanCfgTmp"
-        chown mailman "$mailmanCfgTmp"
+        chown mailman:mailman "$mailmanCfgTmp"
```
Comment (alyssais) [inline nixos/modules/services/mail/mailman.nix]: As above.

File: nixos/modules/services/mail/mailman.nix (hunk 2 above)
Comment (benley) [reply nixos/modules/services/mail/mailman.nix]: Oh you're right - this file should not be shared. I'll revert this part.

File: nixos/modules/services/mail/mailman.nix (hunk 3 above)
Comment (benley) [reply nixos/modules/services/mail/mailman.nix]: Fixing, as above.

File: nixos/modules/services/mail/mailman.nix (hunk 1 above)
Comment (benley) [reply nixos/modules/services/mail/mailman.nix]: Oof sorry, I guess I forgot to include that part. Fixing.

Comment (benley) [reply nixos/modules/services/mail/mailman.nix]: Also for what it's worth this isn't the _only_ way to do this.  Another workaround would be to set mod 0711 on the directories, and make some subparts of them world-readable, but using membership in a shared group seems slightly cleaner to me.

**Comment by Mic92** [comment]
Mailmain has a group in nixpkgs now from what I can see.

**Pull request closed without merging**